drivers/sensor/lsm6dsl/lsm6dsl.c

```diff
@@ -92,7 +92,7 @@ static inline int lsm6dsl_reboot(struct device *dev)
 {
 	struct lsm6dsl_data *data = dev->driver_data;
 
-	if (data->hw_tf->update_reg(data, LSM6DSL_REG_CTRL3_C,
+	if (data->hw_tf->update_reg(dev, LSM6DSL_REG_CTRL3_C,
```

Comment:
Can you explain the need for this commit? What do you mean with "a follow up PR"?

Reply:
Clarified as:

"""
A `struct device` gives access to the runtime per-instance data and the
    constant per-instance configuration.  The per-instance configuration
    will be used later in this series to access to I2C address and SPI
    configuration.
"""